File: LeetCode/hash/IntegerToRoman.java
package hash;

public class IntegerToRoman {

    public String intToRoman(int num) {
        String[] romanSymbols = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int[] values = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};

        StringBuilder roman = new StringBuilder();

        for (int i = 0; i < values.length; i++) {
            while (num >= values[i]) {
                roman.append(romanSymbols[i]);
                num -= values[i];
            }
        }

        return roman.toString();
    }

    public static void main(String[] args) {
        IntegerToRoman solution = new IntegerToRoman();

        System.out.println(solution.intToRoman(3));    // Output: "III"
        System.out.println(solution.intToRoman(4));    // Output: "IV"
        System.out.println(solution.intToRoman(9));    // Output: "IX"
        System.out.println(solution.intToRoman(58));   // Output: "LVIII"
        System.out.println(solution.intToRoman(1994)); // Output: "MCMXCIV"
    }
}

File: LeetCode_Python/hash/IntegerToRoman.py
class IntegerToRoman:
    def intToRoman(self, num: int) -> str:
        roman_numerals = [
            (1000, "M"),
            (900, "CM"),
            (500, "D"),
            (400, "CD"),
            (100, "C"),
            (90, "XC"),
            (50, "L"),
            (40, "XL"),
            (10, "X"),
            (9, "IX"),
            (5, "V"),
            (4, "IV"),
            (1, "I"),
        ]

        res = ""
        for value, symbol in roman_numerals:
            while num >= value:
                res += symbol
                num -= value

        return res


if __name__ == "__main__":
    solution = IntegerToRoman()

    print("Expected Output : MMMDCCXLIX")
    print("Actual Output : ", solution.intToRoman(3749))

    print("Expected Output : LVIII")
    print("Actual Output : ", solution.intToRoman(58))

    print("Expected Output : MCMXCIV")
    print("Actual Output : ", solution.intToRoman(1994))
